lib/pubsub/subscription.js

```diff
@@ -296,32 +300,53 @@ Subscription.prototype.delete = function(callback) {
  *   // message.id = ID used to acknowledge its receival.
  *   // message.data = Contents of the message.
  * });
+ *
+ * //-
+ * // Limit the results.
+ * //-
+ * subscription.pull({ maxCount: 3 }, function(err, messages) {});
  */
 Subscription.prototype.pull = function(options, callback) {
   var that = this;
-  // TODO(jbd): Should not be racing with other pull.
+
   if (!callback) {
     callback = options;
     options = {};
   }
+
+  var batch = util.is(options.maxCount, 'number');
+  var apiEndpoint = batch ? 'subscriptions/pullBatch' : 'subscriptions/pull';
+
   var body = {
     subscription: this.name,
     returnImmediately: !!options.returnImmediately
   };
-  this.makeReq_(
-      'POST', 'subscriptions/pull', null, body, function(err, message) {
-    // TODO(jbd): Fix API to return a list of messages.
+
+  if (batch) {
+    body.maxEvents = options.maxCount;
+  }
+
+  this.makeReq_('POST', apiEndpoint, null, body, function(err, response) {
     if (err) {
       callback(err);
       return;
     }
-    message = Subscription.formatMessage_(message);
+
+    var messages = response.pullResponses || [response];
+    messages = messages.map(Subscription.formatMessage_);
+
+    var messageResponse = response.pullResponses ? messages : messages[0];
+
     if (that.autoAck) {
-      that.ack(message.id, function(err) {
-        callback(err, message);
+      var ackIds = messages.map(function(message) {
+        return message.id;
```

Comment:
Now the message contains messageId, which is different from ackId. The messageId can be used for tracking particular messages.

Can I still access messageId?